src/overlays/actors/ovl_En_Bji_01/z_en_bji_01.c

```diff
@@ -21,36 +48,423 @@ const ActorInit En_Bji_01_InitVars = {
+
+    switch (gSaveContext.entranceIndex) {
+        case 0x4C00: /* Observatory from ECT */
+        case 0x4C10: /* Observatory from Termina Field door */
+            this->actor.params = 0;
+            func_809CCE98(this, globalCtx);
+            break;
+        case 0x4C20: /* Observatory from Termina Field telescope */
+            this->actor.flags |= 0x10000;
+            func_801A5BD0(0);
+            func_801A89A8(0xE0000100);
+            this->actor.params = 3;
+            func_809CCE98(this, globalCtx);
+            break;
+        default:
+            Actor_MarkForDeath(&this->actor);
+            break;
+    }
+}
+
+void EnBji01_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+
+    EnBji01* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnBji01_Update(Actor* thisx, GlobalContext* globalCtx) {
+
+    static s16 sBlinkSequence[6] = { 0, 1, 2, 1, 0, 0 };
+    EnBji01* this = THIS;
+
+    s32 pad; /*pad until 4 main functions take gamestate that gets promoted to globalctx*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CCE98.asm")
+    this->actionFunc(this, globalCtx);
+    Actor_UpdateBgCheckInfo(globalCtx, (Actor*)this, 0.0f, 0.0f, 0.0f, 4U);
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CCEE8.asm")
+    if (this->blinkTimer-- <= 0) {
+        if (--this->blinkSeqIndex < 0) {
+            this->blinkSeqIndex = 4;
+            this->blinkTimer = (Rand_ZeroOne() * 60.0f) + 20.0f;
+        } else {
+            this->eyeTexIndex = sBlinkSequence[this->blinkSeqIndex];
+        }
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD028.asm")
+    Actor_SetHeight(&this->actor, 40.0f);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD328.asm")
+s32 EnBji01_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             Actor* thisx) {
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD634.asm")
+    EnBji01* this = THIS;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD6B0.asm")
+    if ((limbIndex == 0) && ((globalCtx->gameplayFrames & 1) != 0)) {
+        *dList = NULL;
+    }
+    if (limbIndex == 0) {
+        rot->x = rot->x;
+        rot->y = rot->y;
+        rot->z = rot->z;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Bji_01_0x809CCDE0/func_809CD6C0.asm")
+    switch (limbIndex) {
+        case 8: /* Torso */
+            rot->x += this->torsoXRotAdj;
+            rot->z += this->torsoZRotAdj;
+            break;
+        case 15: /* Head */
```

Comment:
You can make an enum for the limbs if you want. There's a few examples in the repo, e.g. Romani.

Reply:
This will be fixed in an upcoming batch commit.